### PR TITLE
Release 158.0.0

### PR DESCRIPTION
**Pull request by mcmire**
## Explanation

The goal of this PR is to release any changes in packages that have not yet been released. This is a prerequisite to merging #3611.

Hence, this PR includes releases for most packages.

## References

As stated above, this is a prerequisite to #3611.

## Changelog

(N/A)

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Comment by mcmire** [comment]
Pinging relevant parties that have code ownership over or have made direct changes to the packages being released here:

- @ccharly @montelaidev (see `accounts-controller`)
- @jiexi (see `polling-controller`)
- @infiniteflower @OGPoyraz (see `preferences-controller`)
- @BZahory (see `user-operation-controller`)

Please look over the changelog entries for the packages mentioned above, cross-reference them against the PRs mentioned in the changelog entries, and let me know if there are any discrepancies.

File: packages/composable-controller/CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +20,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - **BREAKING:** The `ComposableController` class is now a generic class that expects one generic argument `ComposableControllerState` ([#3952](https://github.com/MetaMask/core/pull/3952)).
   - **BREAKING:** For the `ComposableController` class to be typed correctly, any of its child controllers that extend `BaseControllerV1` must have an overridden `name` property that is defined using the `as const` assertion.
+- Bump `@metamask/json-rpc-engine` to `^8.0.2` ([#4234](https://github.com/MetaMask/core/pull/4234))
```
Comment (MajorLift) [inline packages/composable-controller/CHANGELOG.md:23]: Think we'll need to add this!

```suggestion
- **BREAKING:** The types `ComposableControllerStateChangeEvent`, `ComposableControllerEvents`, `ComposableControllerMessenger` are now generic types that expect one generic argument `ComposableControllerState` ([#3952](https://github.com/MetaMask/core/pull/3952)).
- Bump `@metamask/json-rpc-engine` to `^8.0.2` ([#4234](https://github.com/MetaMask/core/pull/4234))
```

**Comment by mcmire** [comment]
Looks like https://github.com/MetaMask/core/pull/4310 was just merged. I'll try to get these added tonight.

File: packages/composable-controller/CHANGELOG.md (hunk 1 above)
Comment (mcmire) [reply packages/composable-controller/CHANGELOG.md:24]: Added in [fc94a75](https://github.com/MetaMask/core/pull/4342/commits/fc94a75bf50ca834bbbebc61d495c317ab907307).

**Comment by mcmire** [comment]
https://github.com/MetaMask/core/pull/4319 was also merged and I will work on incorporating this as well.

**Comment by legobeat** [comment]
Do you think we could consider #4335 for this as well?

**Comment by mcmire** [comment]
@legobeat Yeah, those changes seem pretty easy to add.

**Review by legobeat — APPROVED** [review]
lgtm

**Comment by dbrans** [comment]
@mcmire would you be willing to include https://github.com/MetaMask/core/pull/4343 which I just merged?

File: packages/assets-controllers/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,83 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased]
 
+## [31.0.0]
+
+### Added
+
+- **BREAKING:** The `NftDetectionController` now takes a `messenger`, which can be used for communication ([#4312](https://github.com/MetaMask/core/pull/4312))
+  - This messenger must allow the following actions `ApprovalController:addRequest`, `NetworkController:getState`, `NetworkController:getNetworkClientById`, and `PreferencesController:getState`, and must allow the events `PreferencesController:stateChange` and `NetworkController:stateChange`
+- Add `NftDetectionControllerMessenger` type ([#4312](https://github.com/MetaMask/core/pull/4312))
+- Add `NftControllerMessenger`, `NftControllerGetStateAction`, `NftControllerActions`, `NftControllerStateChangeEvent`, and `NftControllerEvents` types ([#4310](https://github.com/MetaMask/core/pull/4310))
```
Comment (MajorLift) [inline packages/assets-controllers/CHANGELOG.md:17]: The `NftControllerMessenger` was added in v9.0.0 via #1173. Looks like it wasn't recorded in the changelog, but [here's](https://github.com/MetaMask/core/pull/1173/files#diff-7086c0b9bbbb670051a9d0a84a0ab229731eecbe408914be00e1f0e00bebd71dR195-R204) where it was introduced.

```suggestion
- Add `NftControllerGetStateAction`, `NftControllerActions`, `NftControllerStateChangeEvent`, and `NftControllerEvents` types ([#4310](https://github.com/MetaMask/core/pull/4310))
```

Comment (mcmire) [reply packages/assets-controllers/CHANGELOG.md:17]: Ah, my bad. Thanks.

**Review by dbrans — DISMISSED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Comment by mcmire** [comment]
Okay I think we're up to date with `main`! Ready for final approval again.

**Review by cryptodev-2s — APPROVED** [review]
LGTM!

**Pull request merged**